agent/compiler/concrete_compiler.go

```diff
@@ -145,22 +145,22 @@ func (c concreteCompiler) Compile(pkg Package, deps []boshmodels.Package) (blobI
 	return uploadedBlobID, digest, nil
 }
 
-func (c concreteCompiler) fetchAndUncompress(pkg Package, targetDir string) error {
+func (c concreteCompiler) fetchAndUncompress(pkg Package, targetDir string) (string, error) {
```

Comment:
I think rather than returning a path from `fetchAndUncompress` it would be better to partially inline this func and have it only handle fetching. The change does make `Compile()` slightly more verbose but it seems less weird (to me) to return a file path from `fetch` than from `fetchAndUncompress`.

Attaching a patch that illustrates what I'm thinking.

[bosh-agent-no-compression-ALT.patch](https://github.com/user-attachments/files/23734206/bosh-agent-no-compression-ALT.patch)